### ISSUE TITLE
Duplicate messages when using local API

### ISSUE THREAD

**Issue by PulzWave** [steveseguin/social_stream#304]
I'm using the standalone app, version v0.1.51.

I'm building an API to interact with the chat, which requires me to use the HTTPS POST feature (endpoint:  https://io.socialstream.ninja/$sessionid)

But I run into an issue where these chat messages are duplicated. I can reproduce this with the API Sandbox, as seen on this screenshot:

![image](https://github.com/user-attachments/assets/6d90d1ce-0e26-4c49-9f18-494ec42dd930)

Am I expecting the wrong thing, am I doing something wrong or is it a bug?

**Comment by PulzWave** [comment]
To add, I get it that it's not an exact duplicate, as one is posted in the YouTube chat, and the other one in the Twitch that, and that the top left chat is basically consolidated. But the way I want to use this API call would be enough to just print it once in the consolidated chat window.

I did not find any API parameter to control whether to send it to a specific chat. I tried "tid" in the JSON body, and "channel" query parameters, but they both did not change the behaviour.

**Comment by steveseguin** [comment]
Is the goal to send messages to Youtube and Twitch?

If you send a message to those sites, and you are capturing messages of those sights, then its expected you'll get both messages from those sites.

You can enable this option to filter all but the first message however:

![image](https://github.com/user-attachments/assets/7f9e3558-3a1a-4ea7-b8ef-ccf32ef89870)

It might work to filter said messages. should work in most cases.

![image](https://github.com/user-attachments/assets/1eb34973-3e76-4b57-a369-b9190461f66b)

**Comment by PulzWave** [comment]
YES! That's exactly the setting I was looking for, lol. I went through the list of options twice, but apparently managed to miss it twice.

Thank you :)

**Comment by steveseguin** [comment]
There's too many options. Apologies.  I use the search bar at the top to look for keywords now instead

**Comment by PulzWave** [comment]
No worries, having an extended amount of features can not become a negative thing. Bought you a coffee instead ^^
